cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -404,13 +404,34 @@ def get_balance(self, pretty_output: bool = True) -> Union[str, float]:
             return f"${balance:,.2f}"
         return balance
 
-    def get_targets(self) -> Dict[str, List[str]]:
-        """Gets a list of available, unavailable, and retired targets.
+    def get_targets(
+        self,
+        simulator: Optional[bool] = None,
+        supports_submit: Optional[bool] = None,
+        supports_submit_qubo: Optional[bool] = None,
+        supports_compile: Optional[bool] = None,
+        available: Optional[bool] = None,
+        retired: Optional[bool] = None,
+    ) -> List[gss.superstaq_client.TargetInfo]:
+        """Gets a list of Superstaq targets along with their status information.
+
+        Args:
+            simulator: Optional flag to restrict the list of targets to (non-) simulators.
+            supports_submit: Optional boolean flag to only return targets that (don't) allow
+                circuit submissions.
+            supports_submit_qubo: Optional boolean flag to only return targets that (don't)
+                allow qubo submissions.
+            supports_compile: Optional boolean flag to return targets that (don't) support
+                circuit compilation.
+            available: Optional boolean flag to only return targets that are (not) available
+                to use.
+            retired: Optional boolean flag to only return targets that are or are not retired.
 
         Returns:
-            A list of Superstaq targets.
+            A list of Superstaq targets (or a filtered set of targets).
         """
-        return self._client.get_targets()["superstaq_targets"]
+        filters = {key: value for key, value in locals().items() if key != "self"}
```

Comment:
this is cool and definitely works, but i'm a little worried that using `locals()` like this is maybe a little likely to lead to confusing errors down the road, whereas writing out the different args isn't \*too\* much extra work (and assuming prior suggestions will only have to happen this once)
```suggestion
        filters = dict(
            simulator=simulator,
            supports_submit=supports_submit,
            supports_submit_qubo=supports_submit_qubo,
            supports_compile=supports_compile,
            available=available,
            retired=retired,
            **kwargs,
        )
```

Reply:
That's a fair point, updated with `Dict()`.